rad/src/operators/string.rs

```diff
@@ -35,6 +38,117 @@ pub fn parse_json_array(input: &RadonString) -> Result<RadonArray, RadError> {
     item.try_into()
 }
 
+fn magic_map_cleaning(input: BTreeMap<String, RadonTypes>) -> RadonTypes {
+    let map_length = input.len();
+
+    match (map_length, input.get("_text")) {
+        (0, _) => {
+            // <tag></tag>
+            RadonString::from("".to_string()).into()
+        }
+        (1, Some(RadonTypes::Array(rad_array))) => {
+            if rad_array.value().len() == 1 {
+                // <tag>Hello</tag>
+                rad_array.value().into_iter().next().unwrap()
+            } else {
+                // <tag>Hello\nWorld</tag>
+                rad_array.clone().into()
+            }
+        }
+        _ => {
+            let mut new_map: BTreeMap<String, RadonTypes> = BTreeMap::new();
+            for k in input.keys() {
+                match input.get(k) {
+                    Some(RadonTypes::Array(rad_array)) if rad_array.value().len() == 1 => {
+                        let value = rad_array.value().into_iter().next().unwrap();
+                        new_map.insert(k.clone(), value);
+                    }
+                    Some(value) => {
+                        new_map.insert(k.clone(), value.clone());
+                    }
+                    None => unreachable!("Invalid map"),
+                }
+            }
+
+            RadonMap::from(new_map).into()
+        }
+    }
+}
+
+pub fn parse_element_map(
```

Comment:
This looks like it should be a private function

```suggestion
fn parse_element_map(
```